src/coreComponents/physicsSolvers/multiphysics/HydrofractureSolver.cpp

```diff
@@ -176,7 +176,8 @@ real64 HydrofractureSolver< POROMECHANICS_SOLVER >::fullyCoupledSolverStep( real
                                                                             int const cycleNumber,
                                                                             DomainPartition & domain )
 {
-  if( cycleNumber == 0 && time_n <= 0 )
+  // for initial fracture initialization 
```

Comment:
initialNewFractureFields must take care of initial fracture created from execute() in SurfaceGenerator class. Otherwise, the duplication of f-m connections can happen afterwards.    However, when there is a geomechanics initialization step, the previous "IF" will rule out stepping inside due to cycleNumber == 0.

Reply:
@CusiniM @paveltomin

Reply:
I think we need a more intricate initialization/initialcondition procedure to handle this cleanly.